server/src/main/java/org/opensearch/cluster/metadata/RepositoryMetadata.java

```diff
@@ -69,14 +72,30 @@ public class RepositoryMetadata implements Writeable {
      * @param settings repository settings
      */
     public RepositoryMetadata(String name, String type, Settings settings) {
-        this(name, type, settings, RepositoryData.UNKNOWN_REPO_GEN, RepositoryData.EMPTY_REPO_GEN);
+        this(name, type, settings, RepositoryData.UNKNOWN_REPO_GEN, RepositoryData.EMPTY_REPO_GEN, false, null);
+    }
+
+    public RepositoryMetadata(String name, String type, Settings settings, Boolean encrypted, CryptoMetadata cryptoMetadata) {
+        this(name, type, settings, RepositoryData.UNKNOWN_REPO_GEN, RepositoryData.EMPTY_REPO_GEN, encrypted, cryptoMetadata);
     }
 
     public RepositoryMetadata(RepositoryMetadata metadata, long generation, long pendingGeneration) {
-        this(metadata.name, metadata.type, metadata.settings, generation, pendingGeneration);
+        this(metadata.name, metadata.type, metadata.settings, generation, pendingGeneration, metadata.encrypted, metadata.cryptoMetadata);
     }
 
     public RepositoryMetadata(String name, String type, Settings settings, long generation, long pendingGeneration) {
+        this(name, type, settings, generation, pendingGeneration, null, null);
+    }
+
+    public RepositoryMetadata(
+        String name,
+        String type,
+        Settings settings,
+        long generation,
+        long pendingGeneration,
+        Boolean encrypted,
+        CryptoMetadata cryptoMetadata
+    ) {
```

Comment:
If cryptoMetadata is null or empty could it deemed that repository is non-encrypted?

Reply:
Yes, it means that repo is non-encrypted. But we don't enter into a case where repository is marked as encrypted but cryptoMetadata is null. There are validations as well as tests to enforce this.

Reply:
> But we don't enter into a case where repository is marked as encrypted but cryptoMetadata is null. There are validations as well as tests to enforce this.

I commented on this elsewhere as well, but if possible it is better to structure the code to make this invalid case impossible, then there's no need to build any validations. Is there a reason you need a separate boolean as opposed to using the null-ness of `cryptoMetadata` to enable/disable encryption?

Reply:
This is done.